### PR TITLE
update for 3.12

### PR DESCRIPTION
**Pull request by tb102122**
*Issue #, https://github.com/aws-samples/amazon-textract-textractor/issues/288

*Description of changes:*
Extend logic to run on python 3.12

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by tb102122** [comment]
@Belval @schadem Can you please let me know when it can be released. This is the only libary blocking my update to new lambda version python 3.12.

**Comment by tb102122** [comment]
> Could you update your PR to not include formatting changes? 
Done

File: textractor/entities/bbox.py [hunk 1]
```diff
@@ -31,17 +31,17 @@ def __init__(self, width: float, height: float):
 class BoundingBox(SpatialObject):
     """
     Represents the bounding box of an object in the format of a dataclass with (x, y, width, height). \
-    By default :class:`BoundingBox` is set to work with denormalized co-ordinates: :math:`x \in [0, docwidth]` and :math:`y \in [0, docheight]`. \
-    Use the as_normalized_dict function to obtain BoundingBox with normalized co-ordinates: :math:`x \in [0, 1]` and :math:`y \in [0, 1]`. \\
+    By default :class:`BoundingBox` is set to work with denormalized co-ordinates: :math:`x in [0, docwidth]` and :math:`y in [0, docheight]`. \
```
Comment (Belval) [inline textractor/entities/bbox.py:34]: Please revert this change as well, as it will break our documentation.

**Review by Belval — CHANGES_REQUESTED** [review]

**Comment by Belval** [comment]
We should also bump Textractor's version but I can take care of it. cc'ing @kmascar

**Comment by tb102122** [comment]
> We should also bump Textractor's version but I can take care of it. cc'ing @kmascar

done

**Comment by tb102122** [comment]
@Belval thanks for the changes let me know if I should do anythin else.

**Comment by Belval** [comment]
We are still verifying that bumpinmg all the dependencies will not cause any issue, in the meantime if you are using Textractor I did partially merge your changes and will deploy them as part of 1.7.5 if everything passes: https://github.com/aws-samples/amazon-textract-textractor/actions/runs/8114500070/job/22180282101

**Comment by Belval** [comment]
> I addit numpy in the root requirments.txt since it is needed for some tests.

Numpy is part of the extras: https://github.com/aws-samples/amazon-textract-textractor/blob/master/extras/dev.txt we don't enforce it as part of the minimum dependencies as most customers don't need it.

**Comment by miah0x41** [comment]
If the intention is not to support later versions of `numpy` beyond `1.21` then it means anything that uses `pandas` or `[dev]` libraries will be restricted to Python 3.10 or below.

Is it possible to at least then document this somewhere?

**Comment by Belval** [comment]
> If the intention is not to support later versions of numpy beyond 1.21

This is not the intention, let me run a few tests without the pin.

**Comment by tb102122** [comment]
> We are still verifying that bumping all the dependencies will not cause any issue, in the meantime if you are using Textractor I did partially merge your changes and will deploy them as part of 1.7.5 if everything passes: https://github.com/aws-samples/amazon-textract-textractor/actions/runs/8114500070/job/22180282101

Sorry I have missed this comment. The build is working now please check if the layer is build correctly.

**Comment by Belval** [comment]
Thank you for that fix, that's very helpful, were you able to find documentation on why yum is no longer available in the lamdba images?

**Comment by tb102122** [comment]
> Thank you for that fix, that's very helpful, were you able to find documentation on why yum is no longer available in the lamdba images?

Yes here you go. 

https://aws.amazon.com/blogs/compute/introducing-the-amazon-linux-2023-runtime-for-aws-lambda/

**Review by Belval — APPROVED** [review]

**Comment by tb102122** [comment]
@Belval Do you like me to take care of any of the warnings?
The failure is due to permissions I assume or?

https://github.com/aws-samples/amazon-textract-textractor/actions/runs/8182712664

**Comment by Belval** [comment]
Those will resolve once the PR is merged, no action needed on your side.

**Comment by Belval** [comment]
Thank you for your contribution.

**Pull request merged**